build/hostlogswindows/installer/livenessprobe/livenessprobe.cpp

```diff
@@ -7,37 +7,87 @@
 #endif
 
 #include <Windows.h>
-#include <tchar.h>
+#include <tlhelp32.h>
+#include <iostream>
 
 #define SUCCESS 0x00000000
-#define FILESYSTEM_WATCHER_FILE_EXISTS 0x00000001
+#define NO_MONAGENT_LAUNCHER_PROCES 0x00000001
+#define FILESYSTEM_WATCHER_FILE_EXISTS 0x00000002
 #define UNEXPECTED_ERROR 0xFFFFFFFF
 
 /*
-  check if the file exists
+*  Function to check if the given file exists
 */
-bool IsFileExists(const wchar_t *const fileName)
+bool IsFileExists(const wchar_t* const fileName)
 {
     DWORD dwAttrib = GetFileAttributes(fileName);
     return dwAttrib != INVALID_FILE_SIZE;
 }
 
+/*
+* Function to check if a process is running based on its full path
+*/ 
+bool IsMonAgentRunning(const wchar_t* executableFullPath) {
```

Comment:
Theoretically this function could be used to check any process. Maybe a more generic name like IsProcessRunning

Reply:
Renamed to IsProcessRunningWithSpecifcPath